In this week's independent project, you will be working as Data Scientist for MTN Cote d'Ivoire, a leading telecom company and you will be solving for the following research question.

Currently MTN Cote d'Ivoire would like to upgrade its technology infrastructure for its mobile users in Ivory Coast. Studying the given dataset, how does MTN Cote d'Ivoire go about the upgrade of its infrastructure strategy within the given cities?

# Importing Pandas and Numpy

In [0]:
# Importing pandas module in order to work with dataframes
import pandas as pd

# Loading Our datasets


In [2]:
# Loading our cells_geo_description dataset 
# Creating DataFrame from the datasets.
#
dt = "cells_geo_description.xlsx"
df = pd.read_excel(dt, sep=";")
df

Column name                                        Description  Format
0         VILLES                                               City  String
1         STATUS                                  In Service or not  String
2   LOCALISATION                                  in ABIDJAN or not  String
3     DECOUPZONE                                  Geographical Zone  String
4       ZONENAME                                       Name of Zone  String
5      LONGITUDE                                          Longitude   Float
6       LATITUDE                                           Latitude   Float
7         REGION                                             Region  String
8           AREA                                               Area  String
9        CELL_ID                                     ID of the cell  String
10     SITE_CODE  Site (there are several cells per site, severa...  String

In [3]:
# Loading our cells_geo dataset 
# Creating DataFrame from the datasets.
# Converting our column names to lowercase characters
#
dt = "cells_geo.csv"
df_geo = pd.read_csv(dt, sep=';', delimiter=None)
df_geo.columns = map(str.lower, df_geo.columns)
df_geo.head(10)

unnamed: 0    villes      status  ... area     cell_id   site_code
0           0    ADJAME  In Service  ...  NaN  fbb2e75af3  0bb492ef17
1           1    ADJAME  In Service  ...  NaN  1502501d78  0bb492ef17
2           2    ADJAME  In Service  ...  NaN  0c2e182a0c  0bb492ef17
3           3   ASSINIE  In Service  ...  NaN  69bdfc69fd  8a1e6ce31f
4           4   ASSINIE  In Service  ...  NaN  7a1029d2ab  8a1e6ce31f
5           5   ASSINIE  In Service  ...  NaN  0b4f56438f  8a1e6ce31f
6           6  KRIKOREA  In Service  ...  NaN  041d2504f5  260dbfedc5
7           7     ABOBO  In Service  ...  NaN  a70c1d263b  56306cc7ca
8           8     ABOBO  In Service  ...  NaN  e2d04ea883  56306cc7ca
9           9     ABOBO         NaN  ...  NaN  ee224cd27e  56306cc7ca

[10 rows x 12 columns]

In [4]:
# Loading our CDR_description dataset 
# Creating DataFrame from the datasets.
#
dt = "CDR_description.xlsx"
df = pd.read_excel(dt, sep=";")
df

Column name                                        Description   Format
0          PRODUCT                                       Voice or SMS   String
1            VALUE                                      Billing price  Integer
2        DATE_TIME               Time in format yyyy-MM-dd hh:mm:ss.0   String
3     CELL_ON_SITE  Which cell in the site was used (not needed here)  Integer
4  DW_A_NUMBER_INT  Anonymized phone number of the person for whic...   String
5  DW_B_NUMBER_INT        Anonymized phone number of the counterparty   String
6        COUNTRY_A                  Country of party A (useless here)   String
7        COUNTRY_B                  Country of party B (useless here)   String
8          CELL_ID                                     ID of the cell   String
9          SITE_ID                                     ID of the SITE   String

In [5]:
# Loading our Telcom_dataset dataset 
# Creating DataFrame from the datasets.
#
dt = "Telcom_dataset.csv"
df1 = pd.read_csv(dt, sep=",", delimiter=None)
print(df1)


     PRODUTC  VALUE               DATETIME  ... COUNTRY_B     CELL_ID     SITE_ID
0      Voice      0  2012-05-06 23:04:37.0  ...     19e2e  /TJNe+Mmtu  /TJNe+Mmtu
1        sms      0  2012-05-06 23:05:25.0  ...     19e2e  /+cKZKIp41  /+cKZKIp41
2        sms     25  2012-05-06 23:05:41.0  ...     19e2e  /u0FSD+ahi  /u0FSD+ahi
3      Voice     56  2012-05-06 23:05:42.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4      Voice     14  2012-05-06 23:06:29.0  ...     19e2e  051KhYDCpv  051KhYDCpv
...      ...    ...                    ...  ...       ...         ...         ...
4996     sms    100  2012-05-07 00:01:54.0  ...     OTHER  0c2371d9bc  1e751d08a4
4997   Voice      1  2012-05-07 00:01:54.0  ...     19e2e  5fb791cf56  65efe2bc62
4998   Voice      0  2012-05-07 00:01:54.0  ...     19e2e  7773627c12  9ef9616350
4999   Voice      0  2012-05-07 00:01:54.0  ...     19e2e  3f4af49d6c  969b8d0e39
5000   Voice      7  2012-05-07 00:01:54.0  ...     19e2e  6f4d9b77a5  ebfc58af70

[5001 rows x 10

In [6]:
# Loading our Telcom_dataset2 dataset 
# Creating DataFrame from the datasets.
#
dt = "Telcom_dataset2.csv"
df2 = pd.read_csv(dt, sep=",", delimiter=None)
df2.head(10)

PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B     CELL_ID     SITE_ID
0     sms      0  2012-05-07 23:02:06.0  ...     19e2e  +854AcBQT2  +854AcBQT2
1   Voice      0  2012-05-07 23:03:44.0  ...     19e2e  +854AcBQT2  +854AcBQT2
2     sms      0  2012-05-07 23:04:06.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
3   Voice     37  2012-05-07 23:04:48.0  ...     19e2e  /xvRrCVKoQ  /xvRrCVKoQ
4     sms     25  2012-05-07 23:04:58.0  ...     19e2e  +bO+qkSonO  +bO+qkSonO
5     sms      0  2012-05-07 23:05:41.0  ...     19e2e  /MMplBlWjh  /MMplBlWjh
6   Voice     41  2012-05-07 23:06:14.0  ...     19e2e  +laSrk7g6q  +laSrk7g6q
7   Voice     11  2012-05-07 23:06:28.0  ...     19e2e  +gySf02tKW  +gySf02tKW
8     sms      0  2012-05-07 23:06:35.0  ...     OTHER  /0FxP3az1x  /0FxP3az1x
9     sms      0  2012-05-07 23:06:50.0  ...     19e2e  0ecjmwPX21  0ecjmwPX21

[10 rows x 10 columns]

In [7]:
# Loading our Telcom_Dataset3 dataset 
# Creating DataFrame from the dataset.
#
dt = "Telcom_dataset3.csv"
df3 = pd.read_csv(dt, sep=",", delimiter=None)
df3


PRODUCT  VALUE              DATE_TIME  ... COUNTRY_B      CELLID     SIET_ID
0      Voice     61  2012-05-08 23:01:28.0  ...     19e2e  +38u2u/rfx  +38u2u/rfx
1        sms      0  2012-05-08 23:01:31.0  ...     19e2e  +Fs4mTvdKx  +Fs4mTvdKx
2      Voice     10  2012-05-08 23:02:59.0  ...     19e2e  +ZaaZozy/+  +ZaaZozy/+
3        sms      0  2012-05-08 23:03:04.0  ...     19e2e  +cNeJzsTp3  +cNeJzsTp3
4      Voice     22  2012-05-08 23:03:16.0  ...     19e2e  /eeMklgyJA  /eeMklgyJA
...      ...    ...                    ...  ...       ...         ...         ...
4996   Voice      0  2012-05-09 00:01:43.0  ...     19e2e  2aa04dcab7  f7d63a29e5
4997   Voice     62  2012-05-09 00:01:43.0  ...     19e2e  7f1f2f889f  b4ddb9e93c
4998   Voice     32  2012-05-09 00:01:43.0  ...     19e2e  a4731e780e  d68260ef50
4999   Voice     30  2012-05-09 00:01:43.0  ...     19e2e  98ed726da2  b4130b54ab
5000   Voice     60  2012-05-09 00:01:43.0  ...     19e2e  1c11a2fcbb  6d70dae094

[5001 rows x 10 columns]

# Merging our datasets


In [8]:
# Most used city for three days
# Creating a list that contains the four datasets
# Merging the four datasets into one dataset.
# Dropping all the rows with NUll values from the dataset
# Getting the count of individual cities to know how many time each city appears in the column.
# Print the first five cities.
#
frames = [df_geo, df1, df2, df3]
df_dataset = pd.concat(frames, axis=1)
x = df_dataset.dropna()
print(x)


      unnamed: 0    villes      status  ... COUNTRY_B      CELLID     SIET_ID
23          23.0     ABOBO  In Service  ...     19e2e  2q0YsSCxbH  2q0YsSCxbH
24          24.0     ABOBO  In Service  ...     19e2e  1MTLUFpdzd  1MTLUFpdzd
25          25.0     ABOBO  In Service  ...     19e2e  0/eYPFomxx  0/eYPFomxx
26          26.0     ABOBO  In Service  ...     19e2e  ++ctND+jGD  ++ctND+jGD
27          27.0     ABOBO  In Service  ...     OTHER  ++ctND+jGD  ++ctND+jGD
...          ...       ...         ...  ...       ...         ...         ...
3968      3968.0  ZUENOULA  In Service  ...     19e2e  f8f987f764  651c6d67d4
3969      3969.0  ZUENOULA  In Service  ...     19e2e  8b33094233  76595a1e95
3970      3970.0  ZUENOULA  In Service  ...     19e2e  e5e1285933  e9639876a8
3972      3972.0  ZUENOULA  In Service  ...     19e2e  846a0ea574  113d5b7f72
3973      3973.0  ZUENOULA  In Service  ...     19e2e  5a7f6cf0fa  2e828e607c

[2604 rows x 42 columns]


# Getting Insights from our new dataset

**1**. Which ones were the most used city for the three days?

In [9]:
# Getting the count of the column villes('cities')
#
result = pd.value_counts(x['villes'])
city_users = result.head()
c_max = result.max()
print(f"The city with the most users appears: {c_max} times\n")
print(f"The Most used cities in 3 days: \n CITY       COUNT\n{city_users}")

The city with the most users appears: 241 times

The Most used cities in 3 days: 
 CITY       COUNT
COCODY      241
ABOBO       211
YOPOUGON    184
MARCORY      96
ADJAME       81
Name: villes, dtype: int64


**2**. **Which region is most used for the 3 days**

In [10]:
# Getting the region that has most users.
#
region = pd.value_counts(x['region'])
print(f"\nThe most used region in 3 days are:\n")
print(f"Region               Count\n\
-------------------------\n{region.head()}")


The most used region in 3 days are:

Region               Count
-------------------------
LAGUNE               1290
BAS SASSANDRA         155
VALLEE DU BANDAMA     101
LACS                   96
SAVANE                 95
Name: region, dtype: int64


**3. Number of users in Abidjan and those outside of Abidjan.**

In [11]:
# Getting the list of cities in a particular region.
c_loc = pd.value_counts(x['localisation'])
print(c_loc)

INTERIEUR    1431
ABIDJAN      1173
Name: localisation, dtype: int64


**4. Number of Users using Text, Voice and Data in COCODY City.**


> Cocody city had the most number of users.



In [0]:
# Getting the Services data for MTN users in COCODY
voice = x[(x['villes'] == 'COCODY') & (x['PRODUTC'] == "Voice")]
data = x[(x['villes'] == 'COCODY') & (x['PRODUTC'] == "data")]
sms = x[(x['villes'] == 'COCODY') & (x['PRODUTC'] == "sms")]

voice_count = pd.value_counts(voice['PRODUTC'])
data_count = pd.value_counts(data['PRODUTC'])
sms_count = pd.value_counts(sms['PRODUTC'])

print("The Number of Users who Accessed MTN Services in \
\n\n       COCODY city\n-----------------------------\n\
        In 3 Days\n-----------------------------")
print(f"Total Voice users: {voice_count}\n\nTotal data users: {data_count}\n\nTotal sms users: {sms_count}\n\n")

The Number of Users who Accessed MTN Services in 

       COCODY city
-----------------------------
        In 3 Days
-----------------------------
Total Voice users: Voice    121
Name: PRODUTC, dtype: int64

Total data users: data    9
Name: PRODUTC, dtype: int64

Total sms users: sms    111
Name: PRODUTC, dtype: int64




**5. Number of Users using Text, Voice and Data in ABOBO City.**

> Abobo city had the Second most number of users.



In [0]:
# Getting the Services data for MTN users in COCODY city.
# Voice
# data
# sms
#
voice = x[(x['villes'] == 'ABOBO') & (x['PRODUTC'] == "Voice")]
data = x[(x['villes'] == 'ABOBO') & (x['PRODUTC'] == "data")]
sms = x[(x['villes'] == 'ABOBO') & (x['PRODUTC'] == "sms")]

voice_count = pd.value_counts(voice['PRODUTC'])
data_count = pd.value_counts(data['PRODUTC'])
sms_count = pd.value_counts(sms['PRODUTC'])

print("The Number of Users who Accessed MTN Services in \
\n       ABOBO city\n-----------------------------\n\
        In 3 Days\n-----------------------------")
print(f"Total Voice users: {voice_count}\n\nTotal data users: {data_count}\n\nTotal sms users: {sms_count}\n\n")

The Number of Users who Accessed MTN Services in 
       ABOBO city
-----------------------------
        In 3 Days
-----------------------------
Total Voice users: Voice    108
Name: PRODUTC, dtype: int64

Total data users: data    11
Name: PRODUTC, dtype: int64

Total sms users: sms    92
Name: PRODUTC, dtype: int64




**6. Sum of Billing Price used for the 3 days.**

Billing Price In Cococdy City.

In [0]:
v_cocody = x[x['villes'] == 'COCODY']
sum_value = v_cocody['VALUE'].sum()
print(f"The sum of the Billing price per day in Cocody city is\n\
------------------------------------------------------\n{sum_value}")

The sum of the Billing price per day in Cocody city is
------------------------------------------------------
VALUE     5990
VALUE    11296
VALUE    13045
dtype: int64


Billing Price In Abobo City.

In [0]:
v_abobo = x[x['villes'] == 'ABOBO']
sum_value = v_abobo['VALUE'].sum()
print(f"The sum of the Billing price per day in Abobo city is\n\
------------------------------------------------------\n{sum_value}")

The sum of the Billing price per day in Abobo city is
------------------------------------------------------
VALUE    3965
VALUE    3612
VALUE    3489
dtype: int64


# Observations

> Cocody City had the most Number of users.
  It had 241 users in total.
  121 users used Voice,
  9 users used data,
  111 users used sms.

> Abobo City had the second most Number of users.
  It had 211 users in total.
  108 users used Voice,
  11 users used data,
  92 users used sms.

> Most Used regions.
  LAGUNE with 1290,
  BAS SASSANDRA with 155,
  VALLEE DU BANDAMA with 101.

> Most users were from outside Abidjan Region
  with Abidjan having : 1173 and INTERIEUR : 1431

> Cocody City had a much greater billing price for the services than Abobo city.
  Cocody Billing Price >>
  Day 1  :   5990,
  Day 2  :  11296,
  Day 3  :  13045.

  Abobo Billing Price >>
  Day 1  :  3965,
  Day 2  :  3612,
  Day 3  :  3489.